packages/compass-connection-import-export/src/components/export-modal.tsx

```diff
@@ -0,0 +1,135 @@
+import React, { useCallback } from 'react';
+import {
+  Banner,
+  Checkbox,
+  css,
+  FormFieldContainer,
+  FormModal,
+  ToastVariant,
+  useToast,
+} from '@mongodb-js/compass-components';
+import { FileInput } from './file-input';
+import { Passphrase } from './passphrase';
+import { SelectTable } from './select-table';
+import type { ImportExportResult } from '../hooks/common';
+import { useExportConnections } from '../hooks/use-export';
+import type { ConnectionInfo } from 'mongodb-data-service';
+import { usePreference } from 'compass-preferences-model';
+
+const TOAST_TIMEOUT_MS = 5000;
+const tableStyles = css({
+  maxHeight: '24vh',
```

Comment:
Two small comments here, feel free to ignore.

This `maxHeight` value makes it so that on the default compass screen size you start already with two scroll bars in the modal:

<img width="500" alt="image" src="https://user-images.githubusercontent.com/5036933/200606232-23ac0f5c-f3a9-4240-9cfc-42a5ec778410.png">


Maybe it's worth reducing it a little so that there is only table that is scrollable in this case?

I'd maybe also suggest dropping these `FormFieldContainer`s here (due to their margin that you can't disable), changing this to `minHeight` instead and making parent container a grid with the following styles:

```css
display: grid;
grid-template-columns: 100%;
grid-template-rows: 1fr repeat(3, max-content);
gap: 24px;
```

This makes this a grid container where table will take as much space as possible (but not less than the minHeight here) and lead to less scroll on bigger screens:

|<img width="1137" alt="image" src="https://user-images.githubusercontent.com/5036933/200604924-a611215d-437f-46ff-bc61-d05ef9d92328.png">|<img width="1137" alt="image" src="https://user-images.githubusercontent.com/5036933/200605113-32b9f5c4-0475-494e-a00e-6a1115c945a9.png">|
|---|---|

Reply:
Yeah, that also seems like what we’d ideally do, I couldn’t get this to work just by following the “instructions” here – if I only change `maxHeight` to `minHeight`, wrap the modal contents in a div with those css properties and remove the FormFieldContainers, it just results in the table taking up its full height (or `maxHeight` if I add one) – this is most likely just because I misunderstood something here, though, because I believe that your screenshots are real :sweat_smile: 

<details><summary>(diff on top of this PR + screenshot)</summary>

```diff
diff --git a/packages/compass-connection-import-export/src/components/export-modal.tsx b/packages/compass-connection-import-export/src/components/export-modal.tsx
index 195c2487b975..284ba2f7a57a 100644
--- a/packages/compass-connection-import-export/src/components/export-modal.tsx
+++ b/packages/compass-connection-import-export/src/components/export-modal.tsx
@@ -3,8 +3,8 @@ import {
   Banner,
   Checkbox,
   css,
-  FormFieldContainer,
   FormModal,
+  spacing,
   ToastVariant,
   useToast,
 } from '@mongodb-js/compass-components';
@@ -18,10 +18,17 @@ import { usePreference } from 'compass-preferences-model';
 
 const TOAST_TIMEOUT_MS = 5000;
 const tableStyles = css({
-  maxHeight: '24vh',
+  minHeight: '20vh',
   overflow: 'auto',
 });
 
+const modalContainerStyles = css({
+  display: 'grid',
+  gridTemplateColumns: '100%',
+  gridTemplateRows: '1fr repeat(3, max-content)',
+  gap: spacing[4]
+})
+
 export function ExportConnectionsModal({
   open,
   setOpen,
@@ -93,6 +100,7 @@ export function ExportConnectionsModal({
       }
       data-testid="connection-export-modal"
     >
+      <div className={modalContainerStyles}>
       <SelectTable
         className={tableStyles}
         items={connectionList}
@@ -100,7 +108,6 @@ export function ExportConnectionsModal({
         disabled={inProgress}
         onChange={onChangeConnectionList}
       />
-      <FormFieldContainer>
         <FileInput
           label="Target File"
           mode="save"
@@ -108,8 +115,6 @@ export function ExportConnectionsModal({
           onChange={onChangeFilename}
           value={filename}
         />
-      </FormFieldContainer>
-      <FormFieldContainer>
         <Checkbox
           checked={removeSecrets}
           disabled={inProgress || protectConnectionStrings}
@@ -118,8 +123,6 @@ export function ExportConnectionsModal({
           onChange={onChangeRemoveSecrets}
           data-testid="connection-export-remove-secrets"
         />
-      </FormFieldContainer>
-      <FormFieldContainer>
         <Passphrase
           label="Encryption Password"
           description="Optional passphrase to encrypt secrets such as passwords and access tokens"
@@ -128,8 +131,8 @@ export function ExportConnectionsModal({
           onChange={onChangePassphrase}
           value={passphrase}
         />
-      </FormFieldContainer>
       {error && <Banner variant="danger">Error: {error}</Banner>}
+      </div>
     </FormModal>
   );
 }
```

![Screenshot from 2022-11-08 17-34-02](https://user-images.githubusercontent.com/899444/200622362-c4202f8c-9d2f-4864-9eb8-742c2a9b4126.png)


</details>

Reply:
Ah, that's on me, I was trying this out just in the browser devtools, not in the source. For this to work you'd need to make the form body into a grid (the result you are seeing is because the form body has overflow auto, which allows the inner div that you added to stretch to full height when browser is composing the layout) and this `FormModal` component doesn't allow any additional class names to be passed. I am assuming it's not allowed for a reason, so tbh not sure we'd want to change this, sorry for the detour!